editor/editor_resource_picker.cpp

```diff
@@ -895,3 +897,42 @@ void EditorScriptPicker::_bind_methods() {
 
 EditorScriptPicker::EditorScriptPicker() {
 }
+
+// EditorShaderPicker
+
+void EditorShaderPicker::set_create_options(Object *p_menu_node) {
+	PopupMenu *menu_node = Object::cast_to<PopupMenu>(p_menu_node);
+	if (!menu_node) {
+		return;
+	}
+
+	menu_node->add_icon_item(get_theme_icon("ScriptCreate", "EditorIcons"), TTR("New Shader"), OBJ_MENU_NEW_SHADER);
```

Comment:
Shouldn't this have Shader icon instead of Script?
Script has separate icons, because it has multiple actions. Here you can only create, so Shader icon is ok.

Reply:
Ah, yeah, it makes sense (VisualShader is a descendant from Shader too)